drgn_tools/kvm.py

```diff
@@ -183,6 +182,70 @@ def print_vcpu_list(prog: Program) -> None:
     print_table(rows)
 
 
+def print_memslot_info(prog: Program) -> None:
+    vm_list = for_each_vm(prog)
+    rows = [
+        [
+            "KVM",
+            "KVM_MEMSLOTS",
+            "KVM_MEMORY_SLOT",
+            "BASE_GFN",
+            "PAGES",
+            "ARCH",
+            "USER_ADDR",
+            "FLAGS",
+        ]
+    ]
+
+    for vm in vm_list:
+        for i in range(len(vm.memslots)):
+            kvm_memslots = hex(vm.memslots[i])
```

Comment:
Each time you do `vm.memslots[i]`, drgn needs to create a new Object to represent it. Unfortunately, Python can't optimize these things like a C compiler could. So it would be much better to write the loop like this:
```
for memslot in vm.memslots:
    kvm_memslots = hex(memslot)
    ...
```
That way you've created the `memslot` object once, and you can use it in place of `vm.memslots[i]` for the body of the loop below. The variable name obviously could be changed - I'm not familiar with this, and my name may not be a very good one.